modules/core/src/test/scala/doobie/util/FragmentsSuite.scala

```diff
@@ -17,276 +17,131 @@ class FragmentsSuite extends munit.FunSuite {
-  test("whereAndOpt (Foldable empty)") {
-    assertEquals(whereAndOpt(List.empty[Fragment]).query[Unit].sql, "")
-  }
-
-  test("whereAndOpt (Foldable many)") {
-    assertEquals(whereAndOpt(fs).query[Unit].sql, "WHERE ? AND ? AND ? ")
-  }
-
-  test("whereOr (varargs single)") {
-    assertEquals(whereOr(fs(0)).query[Unit].sql, "WHERE ? ")
-  }
 
-  test("whereOr (varargs many)") {
-    assertEquals(whereOr(fs(0), fs(0), fs(0)).query[Unit].sql, "WHERE ? OR ? OR ? ")
+  test("whereOr (many)") {
+    assertEquals(whereOr(fs).query[Unit].sql, "WHERE (? OR ? OR ? ) ")
   }
 
-  test("whereOr (Reducible 1)") {
-    assertEquals(whereOr(nel1).query[Unit].sql, "WHERE ? ")
+  test("whereOr (single)") {
+    assertEquals(whereOr(fs(0)).query[Unit].sql, "WHERE (? ) ")
   }
 
-  test("whereOr (Reducible many)") {
-    assertEquals(whereOr(nel).query[Unit].sql, "WHERE ? OR ? OR ? ")
+  test("whereOr (empty)") {
+    assertEquals(whereOr(List[Fragment]()).query[Unit].sql, "")
   }
 
-  test("whereOrOpt (varargs many Some)") {
-    assertEquals(whereOrOpt(someF, someF).query[Unit].sql, "WHERE ? OR ? ")
+  test("whereOrOpt (many)") {
+    assertEquals(whereOrOpt(ofs: _*).query[Unit].sql, "WHERE (? OR ? ) ")
   }
 
-  test("whereOrOpt (varargs 1 Some)") {
-    assertEquals(whereOrOpt(ofs(0)).query[Unit].sql, "WHERE ? ")
+  test("whereOrOpt (one)") {
+    assertEquals(whereOrOpt(ofs(0)).query[Unit].sql, "WHERE (? ) ")
   }
 
-  test("whereOrOpt (varargs all none)") {
+  test("whereOrOpt (none)") {
     assertEquals(whereOrOpt(None, None).query[Unit].sql, "")
   }
 
-  test("whereOrOpt (Foldable empty)") {
-    assertEquals(whereOrOpt(List.empty[Fragment]).query[Unit].sql, "")
-  }
-
-  test("whereOrOpt (Foldable many)") {
-    assertEquals(whereOrOpt(fs).query[Unit].sql, "WHERE ? OR ? OR ? ")
-  }
-  
-  test("orderBy (varargs 1)") {
-    assertEquals(orderBy(fr0"a").query[Unit].sql, "ORDER BY a")
-  }
-
-  test("orderBy (varargs many)") {
-    assertEquals(orderBy(fr0"a", fr0"b").query[Unit].sql, "ORDER BY a, b")
-  }
-
-  test("orderBy (Reducible 1)") {
-    assertEquals(orderBy(NonEmptyList.of(fr0"a")).query[Unit].sql, "ORDER BY a")
-  }
-
-  test("orderBy (Reducible many)") {
-    assertEquals(orderBy(NonEmptyList.of(fr0"a", fr0"b")).query[Unit].sql, "ORDER BY a, b")
-  }
-
-  test("orderByOpt (varargs Some many) ") {
-    assertEquals(orderByOpt(Some(fr0"a"), Some(fr0"b")).query[Unit].sql, "ORDER BY a, b")
-  }
-
-  test("orderByOpt (varargs all None) ") {
-    assertEquals(orderByOpt(None, None).query[Unit].sql, "")
-  }
-
-  test("orderByOpt (Foldable empty) ") {
-    assertEquals(orderByOpt(List.empty[Fragment]).query[Unit].sql, "")
-  }
-
-  test("orderByOpt (Foldable many) ") {
-    assertEquals(orderByOpt(List(fr0"a", fr0"b")).query[Unit].sql, "ORDER BY a, b")
-  }
-  
-  test("Usage test: whereAndOpt") {
-    assertEquals(whereAndOpt(Some(fr"hi"), orOpt(List.empty[Fragment]), orOpt(List(fr"a", fr"b"))).query[Unit].sql, "WHERE hi AND (a OR b ) ")
```

Comment:
Do you think given this example is painful to work with? I feel like this is alright?

Reply:
Or course, this small example is fine as such. But in more complex queries, the options are more annoying to work with.

With this PR taken into account, you can express the same idea like this:
```scala
whereAnd(Some(fr"hi"), orOpt(List.empty[Option[Fragment]], onEmpty = true), or(List(fr"a", fr"b")))
```